### ISSUE TITLE
Form1.cs is unsaved status after reopening the form designer

### ISSUE THREAD

**Issue by Ashley-Li** [dotnet/winforms#8579]
### .NET version

.NET 8.0.100-preview.2.23106.11

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

Yes, it cannot repro on .NET 6.0 & 7.0.

### Issue description

When save all project, close the form designer, then reopen the form designer, the Form1.cs[Design] is unsaved state.
![image](https://user-images.githubusercontent.com/56664234/217184220-c6a8d1ab-02c0-447a-99b8-4657d1b48bcd.png)

### Steps to reproduce

1.  Create a WinForms .NET 8.0 project.
2.  Open the Smart Tag window of DataGridView, expand "Data Source" comboBox to click "Add new Object Data Source…"
3.  In "Add/Update Datasource Type" dialog, choose a DataSource, e.g. CSharpCodeProvider to bind.
4.  Save All(Ctrl+Shift+S), close the form designer, then reopen the form designer.
5.  Repeat step4 twice to observe.

Or

1.  Extract "[TestApp.zip](https://github.com/dotnet/winforms/files/10672691/TestApp.zip)“ and open it in VS.
2.  Save All(Ctrl+Shift+S), close the form designer, then reopen the form designer.

**Actual result:**
The Form1.cs[Design] is unsaved state, after step4, a dialog box pops up to save the Form1.cs file.
![issue_8 0_actual](https://user-images.githubusercontent.com/56664234/217182045-ac5fbe23-f9da-4573-9f0f-28c58478377d.gif)

**Expected result:**
The Form1.cs[Design] should be saved state, after step4, there's no dialog box pops up to save the Form1.cs file.
![issue_7 0_expected](https://user-images.githubusercontent.com/56664234/217182013-ef1adb77-cce3-4be9-9d7d-b412e32fbd76.gif)

**Comment by dreddy-work** [comment]
@Ashley-Li, can you please open an issue in the designer repo for this and close it here?

**Comment by Ashley-Li** [comment]
@dreddy-work Have filed new issue [4921](https://github.com/microsoft/winforms-designer/issues/4921) in the designer repo. Closing this,
